OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/optimobile/PushRegistration.java

```diff
@@ -57,6 +66,41 @@ public void run() {
             }
         }
 
+        private void requestPermissionIfNeeded(Context context){
+            // TODO: handle multiple requests / app lifecycle changes
+
+            if (Build.VERSION.SDK_INT < 33 || context.checkSelfPermission(Manifest.permission.POST_NOTIFICATIONS) == PackageManager.PERMISSION_GRANTED) {
+                return;
+            }
+
+            Optimobile.handler.post(() -> OptimobileInitProvider.getAppStateWatcher()
+                    .registerListener(new AppStateWatcher.AppStateChangedListener() {
+                        @Override
+                        public void appEnteredForeground() {
+
+                        }
+
+                        @Override
+                        public void activityAvailable(@NonNull Activity activity) {
```

Comment:
does it mean we request the permission only when a new activity is started?

Reply:
Starting an Activity requires an Activity context to launch from, hence use of listener.